### DISCUSSION TITLE
How does onRecordBeforeUpdateRequest work?

### DISCUSSION THREAD

**Discussion by Doctorthe113** [pocketbase/pocketbase#5823] [Q&A] [answered]
If i understand the docs correctly then this code block should (first)console.log how the record was before it was updated.
```js
onRecordBeforeUpdateRequest((e) => {
    console.log("\x1b[32m%s\x1b[0m", JSON.stringify(e.record));

    // This is supposed to grab the updated record, right?
    const userNewRecord = $app.dao().findRecordById("users", userId);
    console.log("\x1b[32m%s\x1b[0m", JSON.stringify(userNewRecord));
}
```
But the `e.record` turns out to be updated record(or atleast how the updated record should look like) and `userNewRecord` is the old, prior to update record. If this is the intended behavior, will `userNewRecord` always be the record prior to receiving update?
I apologize if it's a stupid question. English isn't my first language on top of that I have reading comprehension issue 😅. I just want to make sure my app works reliably.

**Comment by ganigeorgiev** [comment, marked as answer]
It is not clear what actually you are asking or trying to do.

You can find details about the hooks in the [hooks documentation](https://pocketbase.io/docs/js-event-hooks/#onrecordbeforeupdaterequest):

> `onRecordBeforeUpdateRequest` hook is triggered before each API Record update request (**after request data load and before model persistence**).

Or in other words, `e.record` is populated with the update changes BUT that changes are not persisted in the database yet, allowing you to further verify or adjust the data that is going to be saved.

`findRecordById` in the above code returns the "old" state because as mentioned earlier the record update is not saved yet (_the save is performed after all other `onRecordBeforeUpdateRequest` hooks complete without error_).

If that's still not clear, try to be more specific about what you are trying to do.

**Reply by Doctorthe113** [reply]
Oh, I see that makes sense. Thanks for explanation. 

Basically I was trying to compare before and after update of a user record. But I thought `e.record` is the old record and `.findRecordById` would grab the updated record. Hence the confusion 😅. Thanks again!

**Reply by ganigeorgiev** [reply]
You can get the "old"/original record data by calling `e.record.originalCopy().get("yourField")` (_with v0.23.0 the method is renamed to just `original()`_). For other record helpers you can check https://pocketbase.io/docs/js-records/#getset-record-fields.

**Reply by Doctorthe113** [reply]
> You can get the "old"/original record data by calling `e.record.originalCopy().get("yourField")` (_with v0.23.0 the method is renamed to just `original()`_).

Oh I see, good to know. Thank you!
